packages/core/lib/dlc/TxBuilder.ts

```diff
@@ -21,56 +22,72 @@ export class DlcTxBuilder {
     readonly dlcAccept: DlcAcceptWithoutSigs,
   ) {}
 
+  public buildFundingTransaction(): Tx {
+    const txBuilder = new BatchDlcTxBuilder([this.dlcOffer], [this.dlcAccept]);
+    return txBuilder.buildFundingTransaction();
+  }
+}
+
+export class BatchDlcTxBuilder {
+  constructor(
+    readonly dlcOffers: DlcOfferV0[],
+    readonly dlcAccepts: DlcAcceptWithoutSigs[],
+  ) {}
+
   public buildFundingTransaction(): Tx {
     const tx = new TxBuilder();
     tx.version = 2;
     tx.locktime = LockTime.zero();
 
-    const multisigScript =
-      Buffer.compare(
-        this.dlcOffer.fundingPubKey,
-        this.dlcAccept.fundingPubKey,
-      ) === -1
-        ? Script.p2msLock(
-            2,
-            this.dlcOffer.fundingPubKey,
-            this.dlcAccept.fundingPubKey,
-          )
-        : Script.p2msLock(
-            2,
-            this.dlcAccept.fundingPubKey,
-            this.dlcOffer.fundingPubKey,
-          );
-    const witScript = Script.p2wshLock(multisigScript);
-
-    const offerInput = this.dlcOffer.offerCollateralSatoshis;
-    const acceptInput = this.dlcAccept.acceptCollateralSatoshis;
-
-    const totalInput = offerInput + acceptInput;
+    if (this.dlcOffers.length !== this.dlcAccepts.length)
+      throw Error('DlcOffers and DlcAccepts must be the same length');
+    if (this.dlcOffers.length === 0) throw Error('DlcOffers must not be empty');
+    if (this.dlcAccepts.length === 0)
+      throw Error('DlcAccepts must not be empty');
+
+    // Ensure all DLC offers and accepts have the same funding inputs
+    this.ensureSameFundingInputs();
+
+    const multisigScripts: Script[] = [];
+    for (let i = 0; i < this.dlcOffers.length; i++) {
+      const offer = this.dlcOffers[i];
+      const accept = this.dlcAccepts[i];
+
+      multisigScripts.push(
+        Buffer.compare(offer.fundingPubKey, accept.fundingPubKey) === -1
+          ? Script.p2msLock(2, offer.fundingPubKey, accept.fundingPubKey)
+          : Script.p2msLock(2, accept.fundingPubKey, offer.fundingPubKey),
+      );
+    }
+
+    const witScripts = multisigScripts.map((multisigScript) =>
+      Script.p2wshLock(multisigScript),
+    );
 
     const finalizer = new DualFundingTxFinalizer(
-      this.dlcOffer.fundingInputs,
-      this.dlcOffer.payoutSPK,
-      this.dlcOffer.changeSPK,
-      this.dlcAccept.fundingInputs,
-      this.dlcAccept.payoutSPK,
-      this.dlcAccept.changeSPK,
-      this.dlcOffer.feeRatePerVb,
+      this.dlcOffers[0].fundingInputs,
+      this.dlcOffers[0].payoutSPK,
+      this.dlcOffers[0].changeSPK,
+      this.dlcAccepts[0].fundingInputs,
+      this.dlcAccepts[0].payoutSPK,
+      this.dlcAccepts[0].changeSPK,
+      this.dlcOffers[0].feeRatePerVb,
```

Comment:
In the future, we should check that all feeRatePerVb across all DlcOffers are the same

Reply:
#209